src/main/scala/centaur/Test.scala

```diff
@@ -119,52 +120,62 @@ object Operations {
 
   /**
     * This method uses a Metadata JSON returned by Cromwell as it's parameter,
-    * and creates a test map. Since there can be many attempts per call,
-    * this method takes the last call attempt, and filters the metadata
-    * key value pairs by those of interest.
+    * and creates a test map. It flattens the nested JSON into a Map[String, JsValue]
+    * and it filters the keys of interest.
     */
-  private def convertMetadataToTestMap(metadataJson: String): Map[String, JsObject] = {
-    val calls = metadataJson.parseJson.asJsObject.fields.get("calls").get.asJsObject
-    val listOfkeys = Seq("runtimeAttributes", "preemptible", "executionStatus")
-
-    val lastCallAttempt = calls.fields map {
-      case (k1, allCallAttempts: JsArray) =>
-        val lastCallAttempt = allCallAttempts.elements.last.asJsObject
-        val filteredCallAttributes = lastCallAttempt.fields collect {
-          case (k2, callAttributes) if listOfkeys.contains(k2) => k2 -> callAttributes
-        }
-        k1 -> JsObject(filteredCallAttributes)
+  private def makeMetadataMap(metadataJson: String): Map[String, JsValue] = {
 
-      case(k, v) => throw new Exception(s"Wrong Js parameter for $k, Call attempts are expected to be in a JsArray")
-    }
+    val flattenedMetadataMap = metadataJson.parseJson.asJsObject.flatten().fields map { case(k, v) => k -> v }
+    val listOfKeys = Set("runtimeAttributes", "preemptible", "executionStatus")
```

Comment:
I like the one-liner! I truly can't think of a better name, so I'm stealing AllowedKeys ;)